From a2a9d3812f966705e83a8c75a50ee20eb9f005d3 Mon Sep 17 00:00:00 2001
From: Gilberto Garcia <gilbertodamian14@gmail.com>
Date: Sat, 20 Apr 2019 23:31:37 -0500
Subject: [PATCH] Register each plugin with a ProtectedBinder The plugins are
 ProtectedModules, so they need to be registered with a ProtectedBinder
 instead of a normal binder


diff --git a/src/main/java/org/bukkit/craftbukkit/PluginModule.java b/src/main/java/org/bukkit/craftbukkit/PluginModule.java
index 5dd312b3..23fce15b 100644
--- a/src/main/java/org/bukkit/craftbukkit/PluginModule.java
+++ b/src/main/java/org/bukkit/craftbukkit/PluginModule.java
@@ -1,6 +1,7 @@
 package org.bukkit.craftbukkit;
 
 import com.google.inject.AbstractModule;
+import me.fixeddev.inject.ProtectedBinder;
 import org.bukkit.plugin.Plugin;
 
 public class PluginModule extends AbstractModule {
@@ -14,7 +15,7 @@ public class PluginModule extends AbstractModule {
     @Override
     protected void configure() {
         for (Plugin plugin : plugins) {
-            install(plugin);
+            ProtectedBinder.newProtectedBinder(binder()).install(plugin);
 
             requestInjection(plugin);
         }
-- 
2.21.0.windows.1

